### PR TITLE
[REF] web: hoot: char_field conversion

### PR DESCRIPTION
**Pull request by fdardenne**
[[REF] web: hoot: char_field conversion](https://github.com/odoo/odoo/pull/152305/commits/f503ad9ab0a9b92826134ee6e1d80022703e90b8) 
-----------

This commit converts the legacy char_field tests to the new test
framework.

task-3705027

[[FIX] web: hoot: return boolean in set handler](https://github.com/odoo/odoo/pull/152305/commits/6131730088a1a41aff38fa4b76d3e67598e10024) 
-----

According to [1] set should:

- Return true to indicate that assignment succeeded.
- if the set() method returns false, and the assignment happened in
  strict-mode code, a TypeError will be thrown.

Before this commit, the set handler did not return anything, which
resulted in the TypeError being thrown.

Now, the set handler does return a boolean.

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy/set

### CODE REVIEW

File: addons/web/static/tests/views/fields/char_field.test.js [hunk 1]
```diff
@@ -0,0 +1,847 @@
...
+    expect(".o_form_status_indicator_buttons:not(.invisible)").toHaveCount(1, {
+        message: "form view is dirty"
+    });
+    def.resolve();
+    expect(["onchange"]).toVerifySteps();
+    await animationFrame();
+    expect(".o_form_status_indicator_buttons:not(.invisible)").toHaveCount(1, {
+        message: "form view is dirty"
+    });
+    expect(["onchange"]).toVerifySteps();
```
Comment (fdardenne) [inline addons/web/static/tests/views/fields/char_field.test.js:846]: This is a behaviour modification of the test. There is now two "onchange" step in this test.

Before
-------
When calling `editInput`, the mechanism is that the input is modified and then the `input` and `change` events are triggered.
This cause one onchange.

Now
----
Now the new helper `await fieldInput("foo").edit("a");` modify the input and press `Enter`.
This cause two onchange:
1) the onchange of the modified line
2)  the empty onchange of the newline cause `edit` press `Enter`

File: addons/web/static/tests/views/fields/char_field.test.js [hunk 2]
```diff
@@ -0,0 +1,842 @@
...
+    ]
+}
+
+class Partner extends models.Model {
+    _name = "res.partner"
+    _inherit = [];
+
+    foo = fields.Char({ string: "Foo", default: "My little Foo Value", searchable: true, trim: true });
+    int_field = fields.Integer({ string: "int_field", sortable: true, searchable: true });
+    p = fields.One2many({ string: "one2many field", relation: "res.partner", searchable: true });
```
Comment (SimonGenin) [inline addons/web/static/tests/views/fields/char_field.test.js:46]: foo and p are not quality names to understand the type of the field

**Comment by SimonGenin** [comment]
@robodoo r+

@fdardenne Couple of notes: 
- Do not hesitate to change strings like "tralala" and stuff.
- We decided to drop a good amount of message in the assertions. Basically, we keep the ones that really brings information. It helps reduce clutter.

**Comment by SimonGenin** [comment]
@robodoo rebase-ff

**Pull request closed without merging**